src/Testcontainers/Builders/TestcontainersCloudEndpointAuthenticationProvider.cs

```diff
@@ -0,0 +1,127 @@
+﻿namespace DotNet.Testcontainers.Builders
+{
+  using System;
+  using System.Text.Json;
+  using DotNet.Testcontainers.Configurations;
+  using DotNet.Testcontainers.Images;
+  using JetBrains.Annotations;
+
+  /// <inheritdoc cref="IDockerRegistryAuthenticationProvider" />
+  [PublicAPI]
+  internal sealed class TestcontainersHostEndpointAuthenticationProvider : DockerEndpointAuthenticationProvider, ICustomConfiguration
```

Comment:
```suggestion
  internal sealed class TestcontainersCloudEndpointAuthenticationProvider : DockerEndpointAuthenticationProvider, ICustomConfiguration
```

To be in line with the file name, right?

Reply:
Yes, this part was not finally done. The names are now aligned.